src/xdpinstaller/scripts/xdp-setup.ps1

```diff
@@ -0,0 +1,226 @@
+
+# Helper to reboot the machine.
+function Uninstall-Failure {
+    Write-Host "Capturing live kernel dump"
+
+    New-Item -ItemType Directory -Force -Path $LogsDir | Out-Null
+    Write-Verbose "$LiveKD -o $LogsDir\xdp.dmp -k $KD -ml -accepteula"
+    & $LiveKD -o $LogsDir\xdp.dmp -k $KD -ml -accepteula
+
+    Write-Host "##vso[task.setvariable variable=NeedsReboot]true"
+    Write-Error "Preparing to reboot machine!"
+}
+
+# Helper to start (with retry) a service.
+function Start-Service-With-Retry($Name) {
+    Write-Verbose "Start-Service $Name"
+    $StartSuccess = $false
+    for ($i=0; $i -lt 10; $i++) {
+        try {
+            Start-Sleep -Milliseconds 100
+            Start-Service $Name
+            $StartSuccess = $true
+            break
+        } catch { }
+    }
+    if ($StartSuccess -eq $false) {
+        Write-Error "Failed to start $Name"
+    }
+}
+
+# Helper to rename (with retry) a network adapter. On WS2022, renames sometimes
+# fail with ERROR_TRANSACTION_NOT_ACTIVE.
+function Rename-NetAdapter-With-Retry($IfDesc, $Name) {
+    Write-Verbose "Rename-NetAdapter $IfDesc $Name"
+    $RenameSuccess = $false
+    for ($i=0; $i -lt 10; $i++) {
+        try {
+            Rename-NetAdapter -InterfaceDescription $IfDesc $Name
+            $RenameSuccess = $true
+            break
+        } catch {
+            Start-Sleep -Milliseconds 100
+        }
+    }
+    if ($RenameSuccess -eq $false) {
+        Write-Error "Failed to rename $Name"
+    }
+}
+
+# Helper wait for a service to stop and then delete it. Callers are responsible
+# making sure the service is already stopped or stopping.
+function Cleanup-Service($Name) {
+    # Wait for the service to stop.
+    $StopSuccess = $false
+    try {
+        for ($i = 0; $i -lt 100; $i++) {
+            if (-not (Get-Service $Name -ErrorAction Ignore) -or
+                (Get-Service $Name).Status -eq "Stopped") {
+                $StopSuccess = $true
+                break;
+            }
+            Start-Sleep -Milliseconds 100
+        }
+        if (!$StopSuccess) {
+            Write-Verbose "$Name failed to stop"
+        }
+    } catch {
+        Write-Verbose "Exception while waiting for $Name to stop"
+    }
+
+    # Delete the service.
+    if (Get-Service $Name -ErrorAction Ignore) {
+        try { sc.exe delete $Name > $null }
+        catch { Write-Verbose "'sc.exe delete $Name' threw exception!" }
+
+        # Wait for the service to be deleted.
+        $DeleteSuccess = $false
+        for ($i = 0; $i -lt 10; $i++) {
+            if (-not (Get-Service $Name -ErrorAction Ignore)) {
+                $DeleteSuccess = $true
+                break;
+            }
+            Start-Sleep -Milliseconds 10
+        }
+        if (!$DeleteSuccess) {
+            Write-Verbose "Failed to clean up $Name!"
+            Uninstall-Failure
+        }
+    }
+}
+
+# Helper to wait for an adapter to start.
+function Wait-For-Adapters($IfDesc, $Count=1, $WaitForUp=$true) {
```

Comment:
AFAICT this function is unused and can be removed.

Reply:
Removed.